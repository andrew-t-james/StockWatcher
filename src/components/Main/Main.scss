.wrapper {
  align-items: center;
  border: 1px solid red;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-width: 960px;
}