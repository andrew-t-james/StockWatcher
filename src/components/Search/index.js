import React, { Component } from 'react';

class Search extends Component {
  constructor() {
    super();

    this.state = {
      userInput: '',
    };
  }

  handleInputChange(e) {
    const { value } = e.target;

    this.setState({
      userInput: value,
    });
  }

  handleSubmit(e) {
    e.preventDefault();
    console.log('working'); // TODO call fetch method here
    e.target.reset();
    this.setState({ userInput: '' });
  }

  render() {
    const { userInput } = this.state;
    return (
      <form onSubmit={e => this.handleSubmit(e)}>
        <input
          type="text"
          onChange={e => this.handleInputChange(e)}
          value={userInput}
          placeholder="Please Enter a symbol..."
        />
        <button>Add</button>
      </form>
    );
  }
}

export default Search;
