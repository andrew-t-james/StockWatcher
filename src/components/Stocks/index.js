import React from 'react';
import StockCard from '../StockCard/';

const Stocks = props => (
  <section className="stocks-container">
    {props.stocks.map(stock => <StockCard key={stock.shortName} {...stock} />)}
  </section>
);

export default Stocks;
