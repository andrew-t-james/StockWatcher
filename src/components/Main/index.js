import React, { Component } from 'react';
import Search from '../Search';
import Stocks from '../Stocks';

class Main extends Component {
  constructor() {
    super();
    this.state = {
      stocks: [
        {
          name: 'Apple Inc.',
          shortName: 'AAPL',
          high: 188.35,
          low: 185.93,
          latestPrice: 185.24,
          change: -1.26,
        },
        {
          name: 'Alphabet Inc.',
          shortName: 'GOOG',
          high: 1177.295,
          low: 1153.83,
          latestPrice: 1161.44,
          change: -8.4,
        },
      ],
    };
  }

  render() {
    const { stocks } = this.state;
    return (
      <main className="wrapper">
        <h1 className="main-header">Stock Watcher</h1>
        <Search />
        <Stocks stocks={stocks} />
      </main>
    );
  }
}

export default Main;
