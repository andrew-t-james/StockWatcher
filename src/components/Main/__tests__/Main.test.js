import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Main from '../index';
import Search from '../../Search';
import Stocks from '../../Stocks';

describe('Main unit test', () => {
  let wrapper;

  beforeEach(() => (wrapper = shallow(<Main />)));
  afterEach(() => wrapper.unmount());

  test('should render children components', () => {
    const search = wrapper.find(Search).length;
    const stocks = wrapper.find(Stocks).length;

    expect(search).toBe(1);
    expect(stocks).toBe(1);
  });

  test('should pass default props to Stocks component', () => {
    const actual = wrapper.find(Stocks).props();
    const expected = { stocks: [] };

    expect(actual).toEqual(expected);
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
