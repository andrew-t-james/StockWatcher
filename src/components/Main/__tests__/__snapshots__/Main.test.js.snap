// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main unit test should match snapshot 1`] = `
<main
  className="wrapper"
>
  <h1
    className="main-header"
  >
    Stock Watcher
  </h1>
  <Search />
  <Stocks
    stocks={
      Array [
        Object {
          "change": 1.26,
          "high": 188.35,
          "latestPrice": 185.24,
          "low": 185.93,
          "name": "Apple Inc.",
          "shortName": "AAPL",
        },
        Object {
          "change": -8.4,
          "high": 1177.295,
          "latestPrice": 1161.44,
          "low": 1153.83,
          "name": "Alphabet Inc.",
          "shortName": "GOOG",
        },
      ]
    }
  />
</main>
`;
